lib/qo/matchers/matcher.rb

```diff
@@ -0,0 +1,298 @@
+module Qo
+  module Matchers
```

Comment:
Combined hash and array matchers back into one file. Slight speed hit, but easier to work with